docs/developers/creating-your-first-plugin.md

```diff
@@ -0,0 +1,103 @@
+
+---
+title: Creating your First Plugin
+---
+
+Writing plugins for Velocity is not complicated. This section will teach you how to setup your IDE, your plugin
+identifiers, and the basics of the API.
+
+## Set up your environment
+
+You're going to need the [JDK](https://adoptopenjdk.net) and an IDE. If you don't have an IDE, IntelliJ IDEA is recommended.
+
+## Creating your IDE project
+
+* Open your IDE
+* Click `Create New Project` or the equivalent
+* Select either `Gradle` or `Maven`
+* Make sure your **Project JDK** is Java 8 or later
+* **Finish** the dialog and open the project.
+
+Now we have created our project, we need configure our build system. 
+
+## I know how to do this. Give me what I need!
+
+### Maven repository
+
+| Name | velocity                                                   |
+|------|------------------------------------------------------------|
+| URL  | https://nexus.velocitypowered.com/repository/maven-public/ |
+
+### Dependency
+
+| Group ID    | com.velocitypowered |
+|-------------|---------------------|
+| Artifact ID | velocity-api        |
+| Version     | 1.1.0-SNAPSHOT      |
+
+### Javadocs
+
+Javadocs are available at [jd.velocitypowered.com](https://jd.velocitypowered.com).
+
+## Setting up your first project
+
+If you need help setting up your project, don't worry!
```

Comment:
This only covers setting up Gradle or Maven dependencies - we can simplify this header structure